web/src/unread.js

```diff
@@ -434,8 +434,15 @@ class UnreadTopicCounter {
 
             const stream_bucketer = this.bucketer.get_bucket(stream_id);
             const topic = stream_bucketer.reverse_lookup.get(message_id);
-            if (user_topics.is_topic_unmuted(stream_id, topic)) {
-                streams_with_unmuted_mentions.add(stream_id);
+            const stream_is_muted = sub_store.get(stream_id)?.is_muted;
+            if (stream_is_muted) {
+                if (user_topics.is_topic_unmuted(stream_id, topic)) {
+                    streams_with_unmuted_mentions.add(stream_id);
+                }
+            } else {
+                if (!user_topics.is_topic_muted(stream_id, topic)) {
+                    streams_with_unmuted_mentions.add(stream_id);
+                }
```

Comment:
This seems to change the definition of this function without changing the definition in the comment at the start of it.

Please don't ever do that, it's really confusing.